great_expectations/datasource/data_connector/configured_asset_filesystem_data_connector.py

```diff
@@ -71,8 +71,11 @@ def _get_data_reference_list_for_asset(self, asset: Optional[Asset]) -> List[str
 
         if asset is not None:
             if asset.base_directory:
-                base_directory = normalize_directory_path(
-                    dir_path=asset.base_directory, root_directory_path=base_directory
+                base_directory = str(
```

Comment:
could we refactor so we could continue using a `Path` object?

Reply:
@cdkini Yes, but I feel that this should be done in a separate effort as it may impact significant amount of V3.